File: server/java/src/main/java/com/worldlinesolutions/onlinepayments/example/payment/CreatePaymentService.java
package com.worldlinesolutions.onlinepayments.example.payment;

import com.onlinepayments.domain.CreatePaymentRequest;
import com.onlinepayments.domain.CreatePaymentResponse;
import com.onlinepayments.merchant.MerchantClientInterface;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Service;

/**
 * Create payment service used for create payment scenarios
 */
@Service
public class CreatePaymentService {

    private static Logger log = LoggerFactory.getLogger(CreatePaymentService.class);

    private final MerchantClientInterface merchantClient;

    public CreatePaymentService(
            MerchantClientInterface merchantClientFromCommunicator
    ) {
        this.merchantClient = merchantClientFromCommunicator;
    }

    /**
     * Creates a payment from hosted tokenization scenario
     * @param createPaymentRequest {@link CreatePaymentRequest} that originates from Hosted Tokenization
     * @return Instance of {@link CreatePaymentResponse}
     */
    public CreatePaymentResponse createHostedTokenizationPayment(CreatePaymentRequest createPaymentRequest) {
        log.info("Creating hosted checkout request for payment {} {} and hosted tokenization id {}",
                createPaymentRequest.getOrder().getAmountOfMoney().getAmount(),
                createPaymentRequest.getOrder().getAmountOfMoney().getCurrencyCode(),
                createPaymentRequest.getHostedTokenizationId()
        );
        CreatePaymentResponse paymentResponse = merchantClient.payments().createPayment(createPaymentRequest);
        log.info("Successful payment with payment id {}", paymentResponse.getPayment().getId());
        return  paymentResponse;
    }

    /**
     * Creates a payment for payment scenario (Basic, 3DS ...)
     * @param createPaymentRequest {@link CreatePaymentRequest} that originates from Create Payment scenarios
     * @return Instance of {@link CreatePaymentResponse}
     */
    public CreatePaymentResponse createPayment(CreatePaymentRequest createPaymentRequest) {
        log.info("Creating payment request for {} {}",
                createPaymentRequest.getOrder().getAmountOfMoney().getAmount(),
                createPaymentRequest.getOrder().getAmountOfMoney().getCurrencyCode()
        );
        CreatePaymentResponse paymentResponse = merchantClient.payments().createPayment(createPaymentRequest);
        log.info("Successful payment with payment id {}", paymentResponse.getPayment().getId());
        return  paymentResponse;
    }
}

File: server/dotnet/Payment/CreatePaymentService.cs
using OnlinePayments.Sdk.Domain;
using OnlinePayments.Sdk.Merchant;

namespace OnlinePayments.Sdk.Example;

/// <summary>
/// Create payment service used for create payment scenarios
/// </summary>
public class CreatePaymentService
{
    private readonly ILogger<CreatePaymentService> _logger;
    private readonly IMerchantClient _merchantClient;

    public CreatePaymentService(
        ILogger<CreatePaymentService> logger,
        MerchantClientConfig merchantClientConfig
    )
    {
        _logger = logger;
        _merchantClient = merchantClientConfig.GetMerchantClient();
    }

    /// <summary>
    /// Creates a payment request for hosted tokenization scenario
    /// </summary>
    /// <param name="createPaymentRequest">Create payment request</param>
    /// <returns>Task of CreatePaymentResponse</returns>
    public async Task<CreatePaymentResponse> CreateHostedTokenizationPayment(CreatePaymentRequest createPaymentRequest) {
        _logger.LogInformation("Creating hosted checkout request for payment {} {} and hosted tokenization id {}",
            createPaymentRequest.Order.AmountOfMoney.Amount,
            createPaymentRequest.Order.AmountOfMoney.CurrencyCode,
            createPaymentRequest.HostedTokenizationId
        );
        var paymentResponse = await _merchantClient.Payments.CreatePayment(createPaymentRequest);
        _logger.LogInformation("Successful payment with payment id {}", paymentResponse.Payment.Id);
        return  paymentResponse;

    }

    /// <summary>
    /// Creates a payment for payment scenario (Basic, 3DS ...)
    /// </summary>
    /// <param name="createPaymentRequest">Create payment request</param>
    /// <returns>Task of Create payment response</returns>
    public async Task<CreatePaymentResponse> CreatePayment(CreatePaymentRequest createPaymentRequest) {
        _logger.LogInformation("Creating payment request {} {}",
            createPaymentRequest.Order.AmountOfMoney.Amount,
            createPaymentRequest.Order.AmountOfMoney.CurrencyCode
        );
        var paymentResponse = await _merchantClient.Payments.CreatePayment(createPaymentRequest);
        _logger.LogInformation("Successful payment with payment id {}", paymentResponse.Payment.Id);
        return  paymentResponse;

    }
}
